File: binarySearch/binarySearch.py
def binarySearch(array, target):
    small = 0
    large = len(array)-1

    while small <= large:
        mid = (small + large)//2
        if array[mid] == target:
            return mid
        elif array[mid] > target:
            large = mid - 1
        elif array[mid] < target:
            small = mid + 1
    return -1

File: binarySearch/binarySearch.swift
func binarySearch(array: [Int], target: Int) -> Int {
  var small = 0
  var large = array.count - 1

  while small <= large {
    let middle = (small + large) / 2
    if target == array[middle] {
      return middle
    } else if target < array[middle] {
      large = middle - 1
    } else {
      small = middle + 1
    }
  }
  
  return -1
}
